rust/noosphere-gateway/src/handlers/v0alpha1/did.rs

```diff
@@ -1,6 +1,22 @@
-use axum::{http::StatusCode, Extension};
-use noosphere_core::data::Did;
+use axum::http::StatusCode;
+use noosphere_core::context::HasMutableSphereContext;
+use noosphere_storage::Storage;
 
-pub async fn did_route(Extension(gateway_identity): Extension<Did>) -> Result<String, StatusCode> {
-    Ok(gateway_identity.into())
+use crate::{middleware::GatewaySphereContext, ContextManager};
+
+pub async fn did_route<M, C, S>(sphere: GatewaySphereContext<M, C, S>) -> Result<String, StatusCode>
+where
+    M: ContextManager<C, S>,
+    C: HasMutableSphereContext<S>,
+    S: Storage + 'static,
+{
+    let did = sphere
+        .sphere_context()
```

Comment:
A lightweight `get_sphere_id(counterpart: &Did)` or something may fit the bill here. Database (postgres) access is inevitable, though we don't need to create a sphere context. We've worked around the "new database connection on request" as its a non-starter for pretty much all DBs

Reply:
We're talking about a static string that is always the same, with a 1:1 mapping to the input (in the cloud case, the value extracted from the `Host` header). We will not need to open a connect to the DB to serve it in a response.

Reply:
Here are some docs on GCP Cloud CDN: https://cloud.google.com/cdn/docs/caching

It should be possible for us to craft a response that can be served by Cloud CDN for most requests. There are probably some other optimizations to consider as well.

Reply:
And just for the sake of observing the balance that needs to be struck here: serving a static string allows us to be "optimal" in both the self-host and cloud case. A self-hosting user cannot be assumed to be running a sophisticated caching layer in front of their node, so we should strive to be efficient in that case where reasonable.

Reply:
* In all instances, gateways will only ever **open** up a single database connection (or e.g. a single connection pool as an implementation detail)
* Setting some cache headers sounds like an excellent optimization, especially if we only need to set some headers from the server. Is this needed for a V1 here?
* In lieu of a cache or cache expiry, are you suggesting that this can resolve *without* a (postgres, no block storage) query?